### DISCUSSION TITLE
Cannot add plugin to Addon Store

### DISCUSSION THREAD

**Discussion by Aypac** [wlcrs/huawei_solar#254] [Setting up the integration] [answered]
### Describe the issue

I'm trying to install the plugin, but already fail in the first step. I assume what I have to do is the following, where I get this error message:
![image](https://user-images.githubusercontent.com/10585956/211377048-3a5bff42-537a-4567-98b8-0fdad296144e.png)
(same for just `https://github.com/wlcrs/huawei_solar.git`. As you can see it worked for another repo just seconds before)

### Bescribe your Huawei Solar Setup

Should not matter for this.

### How do you connect to the inverter?

Via the SDongle, wireless connection

### Upload your Diagnostics File

I cannot install the app, so I cannot get this diagnostics file.

### Upload your relevant debug logs

```text
2023-01-09 19:05:57.816 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /store/repositories - https://github.com/wlcrs/huawei_solar is not a valid add-on repository
```

### Please confirm the following:

- [X] The problem is still present in the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [X] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I increased the logging configuration, restarted HA, and have copied all relevant log lines from the 'Full Logs' into the textarea.

**Comment by Aypac** [comment, marked as answer]
Ok, I think I figured out the problem.
Rather than just trying to add it where I did, you should install [HACS](https://hacs.xyz/docs/setup/prerequisites) and then add it there.
It took me rather long to figure that out. Is that the suggested method? That was very unclear to me! Would it be worth for me to create a PR with this addition to the documentation?

**Reply by wlcrs** [reply]
I'm open for clarifications to the Readme, but prefer to keep it short & sweet. Otherwise people start to skip/skim the instructions instead of actually reading them. What do you think is lacking to the current instructions?

**Reply by Aypac** [reply]
@g3r0nim00 ; Fair enough. Reading it in hindsight it is actually rather clear. I'm a very new user to Home assistant, so I didn't know what HACS is (I guess I was a bit tired and just assumed that was the regular way of installing packages). Since adding other git repos the way I tried I assumed that was the way to go. Since access to the file structure is a bit tricky with HA, it was not immediately clear to me how to do that either.
@wlcrs / @g3r0nim00 ; I guess the readme is clear for users that have a little more experience with HA and indeed making it longer could be detrimental for those users. How about leaving the readme as is, except for adding a link to a wiki page with more detailed instructions for noobs like me? I'd be willing to set it up/make an according PR.

**Reply by wlcrs** [reply]
Sounds good. Feel free to make a Wiki entry, I'll link it in the Readme once finished

**Reply by Aypac** [reply]
Here's my write-up: https://github.com/wlcrs/huawei_solar/wiki/Detailed-installation-instructions

I tried to avoid duplication with your instructions. I have not tested the manual installation method and wrote up the section for the HACS installation from memory. I will test them in a few days (do not have access to my HA before). I will ping you when I did.

The statement: "Using HACS has the advantage that it can automatically update this integration in the future, while this needs to also be done manually for the second method." I just assumed. Is it true?

**Comment by g3r0nim00** [comment]
https://github.com/wlcrs/huawei_solar

Installation

Install this integration with **HACS**, or copy the contents of this repository into the custom_components/huawei_solar directory
